spec/validator_spec.js

```diff
@@ -20,183 +20,212 @@ function validate(xmlData, error, line = 1) {
-        validate("rootNode", {
+        validateIgnoringNS("rootNode", {
             InvalidChar: "char 'r' is not expected."
         });
     });
 
     it("should validate self closing tags", function () {
-        validate("<rootNode><validtag1  /><validtag2/><validtag3  with='attrib'/><validtag4 />text<validtag5/>text</rootNode>");
+        validateIgnoringNS("<rootNode><validtag1  /><validtag2/><validtag3  with='attrib'/><validtag4 />text<validtag5/>text</rootNode>");
     });
 
     it("should not consider these as self closing tags", function () {
-        validate("<rootNode><validtag1/><invalid tag/><validtag3  with='attrib'/></rootNode>", {
+        validateIgnoringNS("<rootNode><validtag1/><invalid tag/><validtag3  with='attrib'/></rootNode>", {
             InvalidAttr: "boolean attribute 'tag' is not allowed."
         });
-        validate("<rootNode><notSelfClosing/ ></rootNode>", {
+        validateIgnoringNS("<rootNode><notSelfClosing/ ></rootNode>", {
             InvalidAttr: "Attribute '/' has no space in starting."
         });
     });
 
 
     it("should not validate xml string when closing tag is different", function () {
-        validate("<rootNode></rootnode>", {
+        validateIgnoringNS("<rootNode></rootnode>", {
             InvalidTag: "Closing tag 'rootNode' is expected inplace of 'rootnode'."
         });
     });
 
     it("should not validate xml string when closing tag is invalid", function () {
-        validate("<rootNode>< /rootnode>", {
+        validateIgnoringNS("<rootNode>< /rootnode>", {
             InvalidTag: "There is an unnecessary space between tag name and backward slash '</ ..'."
         });
-        validate("<rootNode></ rootnode>", {
+        validateIgnoringNS("<rootNode></ rootnode>", {
             InvalidTag: "There is an unnecessary space between tag name and backward slash '</ ..'."
         });
-        validate("<rootNode></rootnode 123>", {
+        validateIgnoringNS("<rootNode></rootnode 123>", {
             InvalidTag: "Closing tag 'rootnode' can't have attributes or invalid starting."
         });
     });
 
     it("should validate simple xml string with namespace", function () {
-        validate("<root:Node></root:Node>");
+        validateWithNS("<root:Node xmlns:root='urn:none'></root:Node>");
+    });
+
+    it("should not validate simple xml string when namespace is not defined", function () {
+        validateWithNS("<root:Node></root:Node>", {
+            InvalidNS: "Namespace prefix 'root' is not defined for tag 'root:Node'"
+        });
+    });
+
+    it("should not validate xml when namespace is defined later", function () {
+        validateWithNS(`<root:Node>
+        <tag xmlns:root="urn:none">
+        </tag>
+        </root:Node>`, {
+            InvalidNS: "Namespace prefix 'root' is not defined for tag 'root:Node'"
+        });
+    });
+
+    it("should not validate simple xml string when multiple namespace prefixes are present", function () {
```

Comment:
In this case `ns:Node` should be considered as node name and `root` as NS prefix.

Reply:
This can be looked upon in 2 ways, which are in the specifications as well:
1. An XML is namespace-well-formed only when all element and attribute names contain 0 or 1 colon (Ref: https://www.w3.org/TR/xml-names/#Conformance)
2. XML Parsers should allow colon (**:**) in names but authors are discouraged to use it. (Ref: https://www.w3.org/TR/2008/REC-xml-20081126/#sec-common-syn) 

Current implementation follows the first one since we are validating namespaces and most validators follow the same. However, we can introduce a parameter for the first one, if there is a requirement. Your thoughts?

Reply:
we can go ahead with a 0 or 1 colon approach.